plutus-chain-index/src/Plutus/ChainIndex/Api.hs

```diff
@@ -2,21 +2,23 @@
 {-# LANGUAGE TypeOperators #-}
 module Plutus.ChainIndex.Api(API, FromHashAPI) where
 
-import           Ledger                  (Datum, DatumHash, MintingPolicy, MintingPolicyHash, Redeemer, RedeemerHash,
-                                          StakeValidator, StakeValidatorHash, TxId, Validator, ValidatorHash)
-import           Ledger.Credential       (Credential)
-import           Ledger.Tx               (ChainIndexTxOut, TxOutRef)
-import           Plutus.ChainIndex.Tx    (ChainIndexTx)
-import           Plutus.ChainIndex.Types (Diagnostics, Page, Tip)
-import           Servant.API             (Get, JSON, NoContent, Post, Put, ReqBody, (:<|>), (:>))
+import           Ledger                       (Datum, DatumHash, MintingPolicy, MintingPolicyHash, Redeemer,
+                                               RedeemerHash, StakeValidator, StakeValidatorHash, TxId, Validator,
+                                               ValidatorHash)
+import           Ledger.Credential            (Credential)
+import           Ledger.Tx                    (ChainIndexTxOut, TxOutRef)
+import           Plutus.ChainIndex.Pagination (Page, PageQuery)
+import           Plutus.ChainIndex.Tx         (ChainIndexTx)
+import           Plutus.ChainIndex.Types      (Diagnostics, Tip)
+import           Servant.API                  (Get, JSON, NoContent, Post, Put, ReqBody, (:<|>), (:>))
 
 type API
     = "healthcheck" :> Get '[JSON] NoContent
     :<|> "from-hash" :> FromHashAPI
     :<|> "tx-out" :> ReqBody '[JSON] TxOutRef :> Post '[JSON] ChainIndexTxOut
     :<|> "tx" :> ReqBody '[JSON] TxId :> Post '[JSON] ChainIndexTx
     :<|> "is-utxo" :> ReqBody '[JSON] TxOutRef :> Post '[JSON] (Tip, Bool)
-    :<|> "utxo-at-address" :> ReqBody '[JSON] Credential :> Post '[JSON] (Tip, Page TxOutRef)
+    :<|> "utxo-at-address" :> ReqBody '[JSON] (PageQuery TxOutRef, Credential) :> Post '[JSON] (Tip, Page TxOutRef)
```

Comment:
Make the query more intuitive

Reply:
data UtxoAtRequest = UtxoAtRequest { credential :: Credential, pageQuery :: Maybe (PageQuery TxOutRef) }

Reply:
Add comment